spec/features/recipes/welcome_page_spec.rb

```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+feature 'User visits' do
+  scenario 'valid page' do
+    visit('/')
+    expect(page).to have_content "We're gonna have some fun creating this app, old pal"
+  end
+end
```

Comment:
1. Please prepare proper description for `feature`. It should describe in simple words the feature we are testing.
2. Please use word `it` instead of `scenario`. Because when you read it you can notice that if feels more natural (look example at the bottom)
3. Please prepare proper description for `it`. It should describe in simple words the environment changes that happen in the given `scenario` block.
4. Keep all the environment changes outside of the `it` block whenever it is possible. The exception would be case where we have more steps in the middle:
  ```ruby
    before do
      visit('/')  
    end

    it 'redirects user to test page' do
      expect(page).to have_content "We're gonna have some fun creating this app, old pal"

      click_link('go to test page') # Good candidate to move out to `before` but since it happens in the middle we keep it here.
      expect(page).to have_content "We're on the test page!"
    end
  ```

So to summarize it should look like this:

```ruby
feature 'User visits welcome page' do
  before { visit('/') }

  it 'shows user welcome message' do
    expect(page).to have_content "We're gonna have some fun creating this app, old pal"
  end
end
```